prusti-encoder/src/encoders/local_def.rs

```diff
@@ -71,44 +71,44 @@ impl TaskEncoder for MirLocalDefEnc {
         }
 
         vir::with_vcx(|vcx| {
-            let data = if let Some(local_def_id) = def_id.as_local() {
-                let body = vcx.body_mut().get_impure_fn_body(local_def_id, substs, caller_def_id);
-                let locals = IndexVec::from_fn_n(|arg: mir::Local| {
-                    let local = vir::vir_format!(vcx, "_{}p", arg.index());
-                    let ty = deps.require_ref::<RustTyPredicatesEnc>(
-                        body.local_decls[arg].ty,
-                    ).unwrap();
-                    mk_local_def(vcx, local, ty)
-                }, body.local_decls.len());
-                MirLocalDefEncOutput {
-                    locals: vcx.alloc(locals),
-                    arg_count: body.arg_count,
+            if let Some(local_def_id) = def_id.as_local() {
+                if !is_trait_fn_without_impl(vcx.tcx(), def_id) {
```

Comment:
this is very nitty, but could we introduce a `is_trait_fn_with_impl` to avoid the double negation? You mostly seem to check `! without_impl`.

Reply:
The block should also be entered if it's not a trait fn at all (i.e. just a normal function or a fn attached to an `impl`). I suppose a refactoring of the form: `!is_trait_fn(...) || is_trait_fn_with_impl(...)` would work, but I'm not sure if that'd be more clear. No preference either way.

Reply:
could this instead be summarized as "is_function_with_body" ?

Reply:
renamed it!